experimental/segmenter/tools/generate_properties.py

```diff
@@ -43,19 +48,45 @@
             for i in range(start, end + 1):
                 ea_props[i] = prop
 
+with open('emoji-data.txt', 'r') as emoji_file:
+    range_codepoint_pattern = r"([0-9A-F]{1,6})\.\.([0-9A-F]{1,6})\s*\;\s*([A-ZA-z_]{2,})"
+    single_codepoint_pattern = r"([0-9A-F]{1,6})\s*\;\s*([A-Za-z_]{2,})"
+
+    for line in emoji_file.readlines():
+        start = end = prop = None
+        if m := re.match(range_codepoint_pattern, line):
+            start = int(m[1], 16)
+            end = int(m[2], 16)
+            prop = m[3]
+        elif m := re.match(single_codepoint_pattern, line):
+            start = end = int(m[1], 16)
+            prop = m[2]
+
+        if prop != "Extended_Pictographic":
+            continue
+
+        # We have a success match, property is Extended_Pictographic, and the
+        # codepoints are in plane 0 & 1. Store their Extended_Pictographic
+        # list.
+        if prop and start < begin_plane2:
+            for i in range(start, end + 1):
+                extended_pictographics[i] = "Y"
```

Comment:
Question: could be elaborate a bit more on using "Y"? Does it mean "yes" or something else?

Reply:
I will change it to boolean.